test/unit/math/test_ad.hpp

```diff
@@ -1489,6 +1554,45 @@ inline void test_matvar_gradient(MatVar&& A_mv, VarMat&& A_vm,
   }
 }
 
+/**
+ * Test that the jacobian for matrices of vars is equal for the
+ *  var matrix when the result is an either a `var_value` with inner Eigen type
+ * or an Eigen matrix with a scalar `var` type.
+ * @tparam MatVar1 An Eigen type inheriting from `EigenBase` that has
+ *  Scalar vars.
+ * @tparam MatVar2 An Eigen type inheriting from `EigenBase` that has
+ *  Scalar vars.
+ * @tparam VarMat1 A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam VarMat2 A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam ResultMatVar The resulting type of applying a function to `A_mv1`
+ * and `A_mv2`.
+ * @tparam ResultVarMat The result type of applying a function to `A_vm1` and
+ *  `A_vm2`.
+ * @param A_mv1 A eigen type of vars.
+ * @param A_mv2 A eigen type of vars.
+ * @param A_vm1 A eigen type of vars.
+ * @param A_vm2 A eigen type of vars.
+ * @param A_mv_f The result of a function from applying `A_mv1` and `A_mv2`.
+ * @param A_vm_f The result of a function from applying `A_vm1` and `A_vm2`
+ */
+template <typename ResultMatVar, typename ResultVarMat, typename MatVar1,
+          typename MatVar2, typename VarMat1, typename VarMat2,
+          require_eigen_t<ResultMatVar>* = nullptr>
+inline void test_matvar_gradient(ResultMatVar& A_mv_f, ResultVarMat& A_vm_f,
+                                 const MatVar1& A_mv1, const MatVar2& A_mv2,
+                                 const VarMat1& A_vm1, const VarMat2& A_vm2) {
+  for (Eigen::Index i = 0; i < A_mv_f.size(); ++i) {
+    A_vm_f.adj()(i) = 1;
+    A_mv_f.adj()(i) = 1;
+    stan::math::grad();
+    expect_near_rel_var("lhs var<Matrix> vs Matrix<var> result", A_vm1, A_mv1);
+    expect_near_rel_var("rhs var<Matrix> vs Matrix<var> result", A_vm2, A_mv2);
```

Comment:
I prefer 'first argument' and 'second argument' to left hand side/right hand side lingo here because both of the variables are arguments (things that are usually on the right hand side of things).

Reply:
Word that makes sense